library/core/src/ptr/non_null.rs

```diff
@@ -1208,6 +1214,36 @@ impl<T: ?Sized> NonNull<T> {
     #[must_use]
     #[stable(feature = "non_null_convenience", since = "1.80.0")]
     #[rustc_const_unstable(feature = "const_align_offset", issue = "90962")]
+    #[ensures(|result| {
```

Comment:
We should consider simplifying this later or at least move it to an auxiliary function.